doors_dashboards/components/scattermap.py

```diff
@@ -166,19 +168,29 @@ def update_general_store_after_station_selection(
                 general_data["collection"] = (
                     collection_name)
             if "groups" not in general_data:
-                group_name = self.feature_handler.get_levels(collection_name)[-1]
+                group_name = self.feature_handler.get_levels(collection_name)[-1] if \
+                    self.feature_handler.get_levels(collection_name) else ""
                 lon = click_data['points'][0]['lon']
                 lat = click_data['points'][0]['lat']
                 p = Point(lon, lat)
                 gdf = self.feature_handler.get_df(collection_name)
                 gdf = gdf[gdf["geometry"].geom_equals(p)]
                 if len(gdf) > 0:
-                    group = gdf.iloc[0][group_name]
-                    general_data["groups"] = {}
-                    general_data["groups"][collection_name] = (
-                     group)
+                    if group_name is "":
+                        text = click_data['points'][0]['text']
+                        general_data["meteogram_data"] = {
+                            'lon': lon,
+                            'lat': lat,
+                            'label': text
+                        }
+                    else:
+                        group = gdf.iloc[0][group_name]
+                        general_data["groups"] = {}
+                        general_data["groups"][collection_name] = (
+                            group)
```

Comment:
```suggestion
                    text = click_data['points'][0]['text']
                    general_data["selected_point"] = {
                        'lon': lon,
                        'lat': lat,
                        'label': text
                    }
                    if group_name:
                        group = gdf.iloc[0][group_name]
                        if "groups" not in general_data:
                            general_data["groups"] = {}
                        general_data["groups"][collection_name] = (group)
```
I do not understand why you update this differently whether groups are provided or not. Why not do both always?
Please note I changed meteogram_data to selected_point, as this way it is not as commited to another component.